File: examples/graalvm/kora-java-graalvm-crud-jdbc/src/main/java/io/koraframework/example/graalvm/crud/jdbc/model/mapper/PetMapper.java
package io.koraframework.example.graalvm.crud.jdbc.model.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;
import io.koraframework.example.graalvm.crud.jdbc.model.dao.PetCategory;
import io.koraframework.example.graalvm.crud.jdbc.model.dao.PetWithCategory;
import io.koraframework.example.graalvm.crud.openapi.server.model.CategoryTO;
import io.koraframework.example.graalvm.crud.openapi.server.model.PetTO;

@Mapper(unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface PetMapper {

    PetTO asDTO(PetWithCategory pet);

    CategoryTO asDTO(PetCategory category);
}

File: examples/kotlin/kora-kotlin-crud/src/main/kotlin/io/koraframework/kotlin/example/crud/model/PetMapper.kt
package io.koraframework.kotlin.example.crud.model

import io.koraframework.example.crud.openapi.http.server.model.CategoryTO
import io.koraframework.example.crud.openapi.http.server.model.PetTO
import io.mcarle.konvert.api.Konverter

@Konverter
interface PetMapper {

    fun petWithCategoryToPetTO(pet: PetWithCategory): PetTO

    fun petCategoryToCategoryTO(category: PetCategory): CategoryTO
}
